File: 프로그래머스/2/42578. 의상/의상.py
# from collections import defaultdict
# def solution(clothes):
#     answer = 1
#     dic = defaultdict(list)
#     for i, j in clothes:
#         dic[j].append(i)
#     for i in dic:
#         answer *= len(dic[i])+1
#     return answer-1

from collections import defaultdict
def solution(clothes):
    answer = 1
    dic = defaultdict(int)
    for i, j in clothes:
        dic[j] += 1
    for i in dic:
        answer *= dic[i]+1
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> hm = new HashMap<String, Integer>();
        for (String[] a : clothes) {
            hm.put(a[1], hm.getOrDefault(a[1], 0)+1);
        }
        for (String key: hm.keySet()) {
			int value = hm.get(key);
            answer *= value+1;
		}
        return answer-1;
    }
}